src/python/pants/option/global_options.py

```diff
@@ -415,6 +421,20 @@ def register_bootstrap_options(cls, register):
     register('--process-execution-cleanup-local-dirs', type=bool, default=True, advanced=True,
              help='Whether or not to cleanup directories used for local process execution '
                   '(primarily useful for e.g. debugging).')
+    register('--process-execution-speculation-timeout-ms', type=int,
+             default=DEFAULT_EXECUTION_OPTIONS.process_execution_speculation_timeout_ms, advanced=True,
+             help='Number of milliseconds to wait before speculating a second request for a slow process. '
+                  ' see `--process-execution-speculation`')
+    register('--process-execution-speculation', choices=['remote_first', 'local_first', 'none'],
```

Comment:
I keep wanting to call this `--process-execution-speculation-strategy`, but then talk myself out of it. So just going to post the idea and see what you think.

The reason I'm thinking this is that it a) makes the idea more clear and b) keeps the namespace `--process-execution-speculation` for further use. I'm not sure if we support sub-subsytems, where `speculation` is a namespace that exists within `process-execution`? If that's the case, we'd want to keep `speculation` for the overall subsystem.

Downside is more typing.

Reply:
I like `-strategy`, will change.